packages/core/src/odata/v2/extract-data-from-one-to-many-link.ts

```diff
@@ -2,9 +2,18 @@
 
 /**
  * Data extractor for one to many links for v2 entity used in [[entityDeserializer]]
+ * It first tries if data.result is an array and returns it.
+ * Then it tires if data itself is an array and return it.
+ * If both are not the case it returns undefined
  * @param data - One to many link response data
  * @returns The content of the one to many link
  */
 export function extractDataFromOneToManyLink(data): any[] | undefined {
-  return data.results;
+  if (data.results && Array.isArray(data.results)) {
```

Comment:
I don't understand why we have this function in the first place...
Maybe you can consider deprecating it, but I guess this is nothing for the quick fix.

Anyways, even for a quick fix, please use the existing response data accessor: https://github.com/SAP/cloud-sdk/blob/master/packages/core/src/odata/v2/request-builder/response-data-accessor.ts

I must have missed this function when applying it everywhere.

Reply:
The data accessor is a v2 thing. We thought there is a clear cut between v2 and v4. But it seems like it is not so strict so I mad now a single method handling both cases. Let me know what you think.